sys/riscv/riscv/machdep.c

```diff
@@ -609,23 +717,31 @@ sendsig(sig_t catcher, ksiginfo_t *ksi, sigset_t *mask)
 	mtx_assert(&psp->ps_mtx, MA_OWNED);
 
 	tf = td->td_frame;
-	onstack = sigonstack(tf->tf_sp);
+#if __has_feature(capabilities)
+	/*
+	 * XXXRW: We make an on-stack determination using the virtual address
+	 * associated with the stack pointer, rather than using the full
+	 * capability.  Should we compare the entire capability...?  Just
+	 * pointer and bounds...?
+	 */
+#endif
```

Comment:
Only a comment in the #if seems odd...

Reply:
In mips the code is different as it has to look at tf_csp instead of tf_sp.  The comment is still germane, so I copied it over from mips.  I put the #if to point out that it as a CHERI-specific comment.  Maybe there's a better way to do this than #if?